### PR TITLE
Implement futures market details and trading history on left sidebar

### PR DESCRIPTION
**Pull request by Xiaolei-Shawn**
## Description

- Add `FuturesMarketTabs` as container for `TradesHistoryTable` and `FuturesMarketTab`
- Implement `FuturesMarketTab`
- Add `activeTabState` in store for sake of persisting tab state when switching market
- Update `Table` to accept customized row/cell/header props so that the content of cell/header in specific column can be aligned to right.
- Adjust `TradesHistoryTable`
  - Center middle column
  - Align right most column to right

## Related issue

[Issue](https://github.com/Kwenta/kwenta/issues/978)

## Motivation and Context

The reason to extend `Table` component is only styling the child component in table won't affect flex of `TableRow/TableCell/TableHeader`. Extending `Table` to support customization seems to be only solution IMO. I am open for other solution too.

## Screenshots (if appropriate):
Left sidebar

### CODE REVIEW

**Review by platschi — APPROVED** [review]
Deployed version works as intended, well done!

**Review by LeifuChen — COMMENTED** [review]
@Xiaolei-Shawn The solution looks fine with me.

I think another possible solution is overwriting the css rule by css selector directly.
e.g. in `TradesHistoryTable.tsx`, add something like below
```css
const TableAlignment = css`
	justify-content: space-between;
	& >div:first-child {
		flex: 60 60 0 !important;
	}
	& >div:nth-child(2) {
		flex: 100 100 0 !important;
		display: flex;
		justify-content: center;
	}
	& >div:last-child {
		flex: 70 70 0 !important;
		justify-content: flex-end;
		padding-right: 20px;
	}
`;

const StyledTable = styled(Table)<{ mobile?: boolean }>`
        ....
	.table-row {
		${TableAlignment}
	}
	.table-body-row {
		${TableAlignment}
		padding: 0;
	}
`;
```
Here is the table layout after the css change.
image

**Comment by Xiaolei-Shawn** [comment]
> @Xiaolei-Shawn The solution looks fine with me.
> 
> I think another possible solution is overwriting the css rule by css selector directly. e.g. in `TradesHistoryTable.tsx`, add something like below
> 
> ```css
> const TableAlignment = css`
> 	justify-content: space-between;
> 	& >div:first-child {
> 		flex: 60 60 0 !important;
> 	}
> 	& >div:nth-child(2) {
> 		flex: 100 100 0 !important;
> 		display: flex;
> 		justify-content: center;
> 	}
> 	& >div:last-child {
> 		flex: 70 70 0 !important;
> 		justify-content: flex-end;
> 		padding-right: 20px;
> 	}
> `;
> 
> const StyledTable = styled(Table)<{ mobile?: boolean }>`
>         ....
> 	.table-row {
> 		${TableAlignment}
> 	}
> 	.table-body-row {
> 		${TableAlignment}
> 		padding: 0;
> 	}
> `;
> ```
> 

@LeifuChen, I like your solution. TBH, if asking me extending `Table` is kinda overkill and I suppose there is an easy way to solve it. In case we're able to style the component without passing custom style to table, it's recommended to do so. One thing I dislike is `!important` which can be error-prone and working implicitly since it's kinda difficult to predict impacts of `important`. In other hand, adding custom row/header/cell props in `Table` component may not be used commonly given that `Styled` is always preferable (like I mentioned before). I can update it to use css selector if we're not considering `!important` a bad practice.

**Comment by Xiaolei-Shawn** [comment]
@LeifuChen, I think using `styled` is less invasive so let's go with that. Have pushed the change to align with your comment.

**Review by LeifuChen — CHANGES_REQUESTED** [review]
@Xiaolei-Shawn Great stuff. Only request a small change on the `NO_VALUE` placeholder. Thanks.

File: sections/futures/FuturesMarketTabs/constants.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const DEFAULT_DATA = [
```
[the remark this answers is not in the document]
Comment (Xiaolei-Shawn) [reply sections/futures/FuturesMarketTabs/constants.ts:1]: Good catch!

**Review by LeifuChen — APPROVED** [review]
@Xiaolei-Shawn Approved. Thanks for your contribution.

**Review by platschi — APPROVED** [review]
Great work!

**Pull request merged**